pulp_rpm/app/advisory.py

```diff
@@ -120,39 +122,50 @@ def resolve_advisory_conflict(previous_advisory, added_advisory):
-            to_remove.append(previous_advisory_pk)
-            to_exclude.append(added_advisory_pk)
-        else:
+    # Prepare results of conditions for easier use.
+    same_dates = previous_updated_date == added_updated_date
+    same_version = previous_updated_version == added_updated_version
+    pkgs_intersection = previous_pkglist.intersection(added_pkglist)
+
+    if same_dates and same_version and pkgs_intersection:
+        if previous_pkglist != added_pkglist:
             raise AdvisoryConflict(_('Incoming and existing advisories have the same id and '
                                      'timestamp but different and intersecting package lists. '
                                      'At least one of them is wrong. '
                                      f'Advisory id: {previous_advisory.id}'))
-    else:
-        if previous_pkglist.intersection(added_pkglist):
-            if previous_updated_date < added_updated_date:
-                to_remove.append(previous_advisory.pk)
-            else:
-                to_exclude.append(added_advisory.pk)
+    elif (not same_dates and not pkgs_intersection) or \
+            (same_dates and not same_version and not pkgs_intersection):
+        raise AdvisoryConflict(_('Incoming and existing advisories have the same id but '
+                                 'different timestamps and intersecting package lists. It is '
+                                 'likely that they are from two different incompatible remote '
+                                 'repositories. E.g. RHELX-repo and RHELY-debuginfo repo. '
+                                 'Ensure that you are adding content for the compatible '
+                                 f'repositories. Advisory id: {previous_advisory.id}'))
+    elif not same_dates and pkgs_intersection:
+        if previous_updated_date < added_updated_date:
+            to_remove.append(previous_advisory.pk)
+        else:
+            to_exclude.append(added_advisory.pk)
+    elif not same_version and pkgs_intersection:
+        if is_previous_version(previous_updated_version, added_updated_version):
+            to_remove.append(previous_advisory.pk)
         else:
-            raise AdvisoryConflict(_('Incoming and existing advisories have the same id but '
-                                     'different timestamps and intersecting package lists. It is '
-                                     'likely that they are from two different incompatible remote '
-                                     'repositories. E.g. RHELX-repo and RHELY-debuginfo repo. '
-                                     'Ensure that you are adding content for the compatible '
-                                     f'repositories. Advisory id: {previous_advisory.id}'))
+            to_exclude.append(added_advisory.pk)
+    elif same_dates and same_version and not pkgs_intersection:
+        merged_advisory = merge_advisories(previous_advisory, added_advisory)
```

Comment:
we need to save a pk of the previous version and pass it late to remove.
Please leave the comment why as well:
```
# previous_advisory is used to copy the object and thus the variable refers to a different object after `merge_advisories` call
previous_advisory_pk = previous_advisory.pk
merged_advisory = merge_advisories(previous_advisory, added_advisory)
```